packages/api/src/incentives/activity-points/upsertAccountActivityPoints.ts

```diff
@@ -21,7 +21,35 @@ export const UpsertAccountActivityPointsLive = Layer.effect(
 
     return (input) =>
       Effect.gen(function* () {
-        yield* Effect.forEach(chunker(input, 10_000), (chunk) => {
+        // Get unique activity IDs from input
+        const uniqueActivityIds = [...new Set(input.map(item => item.activityId))];
+        
+        // Validate that all activity IDs exist in the activities table
+        const validActivityIds = yield* Effect.tryPromise({
+          try: async () => {
+            const result = await db
+              .select({ id: activities.id })
+              .from(activities)
```

Comment:
use activityWeek table instead, this will give you all the activities that are included for a given weekId